### DISCUSSION TITLE
Understanding requirement of "go2rtc"

### DISCUSSION THREAD

**Discussion by ChrisSutton123** [blakeblackshear/frigate#5407] [General]
I don't understand the advantages of using "go2rtc".
I have two configurations for my REOLINK RLC-520
Both work fine
First one not using "go2rtc"
--------------------------------------------------------------------------
go2rtc:
  streams: {}

cameras:
  RLC-520:
    ffmpeg:
      inputs:
       # 2560x1920
         - path: http://192.168.3.201/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=user&password=password
           input_args: preset-http-reolink         
           roles:
             - record
        # 896x672      
         - path: http://192.168.3.201/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=user&password=password
           input_args: preset-http-reolink
           roles:
             - detect
--------------------------------------------------------------------------
Second one using "go2rtc"
--------------------------------------------------------------------------
 go2rtc:
  streams:
    reolink:
      # 2560x1920 
      - http://192.168.3.201/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=user&password=password
      - ffmpeg:reolink
    reolink_sub:
      # 896x672  
      - http://192.168.3.201/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=user&password=password

cameras:
  RLC-520:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/reolink?video=copy&audio=aac
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/reolink_sub?video=copy
          input_args: preset-rtsp-restream
          roles:
            - detect
----------------------------------------------------------------------------------------------- 

If I am not restreaming the image is the first option better to use?

**Comment by NickM-27** [comment]
We highly recommend setting up go2rtc so that frigate has better live view options in the frontend. It is not required to use go2rtc restream as a source for frigate. You can add go2rtc section and leave the cameras section pointing at the camera (basically a mix of the two options you put)

using the restream as a source for frigate (as you did in second example) is not necessary but can be useful to reduce the number of connections to your camera making it not work as hard.

**Reply by ChrisSutton123** [reply]
Thank you for the explanation
